crates/saya/core/src/prover/client.rs

```diff
@@ -35,14 +34,51 @@ pub async fn http_prove(
 ) -> anyhow::Result<String> {
     let prover =
         ProverSDK::new(prover_params.prover_url.clone(), prover_params.prover_key.clone()).await;
-    let prover = prover.as_ref().map_err(|e| anyhow::anyhow!(e.to_string()))?;
+    let prover = prover.    as_ref().map_err(|e| anyhow::anyhow!(e.to_string()))?;
 
     // TODO: cairo0 might be deprectated in the future.
     if prove_program.cairo_version() == FieldElement::ONE {
         let input = prepare_input_cairo1(input, prove_program).await?;
-        prover.prove_cairo(input).await.context("Failed to prove using the http prover")
+        let job_id = prover.prove_cairo(input).await.context("Failed to prove using the http prover")?;
+        prover.sse(job_id).await?;
+        let response = prover.get_job(job_id).await?;
+        let response = response.text().await?;
+        let json_response: Value = serde_json::from_str(&response)?;
+        if let Some(status) = json_response.get("status").and_then(Value::as_str) {
+            if status == "Completed" {
+                return Ok(json_response
+                    .get("result")
+                    .and_then(Value::as_str)
+                    .unwrap_or("No result found")
+                    .to_string());
```

Comment:
We might have better maintainability working with rust `serde` directly instead of parsing plain JSON though.